strax/context.py

```diff
@@ -381,6 +382,45 @@ def get_components(self, run_id: str,
 
         plugins = self._get_plugins(targets, run_id)
 
+        n_range = None
+        if time_range is not None:
+            # Ensure we have one data kind
+            _kinds = set([plugins[t].data_kind for t in targets])
+            if len(_kinds) > 1:
+                raise NotImplementedError(
+                    "Time range selection not implemented "
+                    "for multiple data kinds.")
+            kind = list(_kinds)[0]
+
+            # Which plugin provides time information? We need it to map to
+            # row indices.
+            for p in targets:
+                if 'time' in plugins[p].dtype.names:
+                    break
+            else:
```

Comment:
Since this is dead code for now, better to just add times to dtypes since this is also useful for the analysis.